### DISCUSSION TITLE
Best approach to get the list values(Array) from a component inside form

### DISCUSSION THREAD

**Discussion by tanetrail** [react-hook-form/react-hook-form#4194] [Q&A]
Similar to FormContext in the below link, Inside a form we have a component which displays/updates list of items. what is the best approach to collect the items from the component during form onSubmit (other than react way like redux store / contextProvider).

https://react-hook-form.com/api/#useFormContext

Got it fixed using "Controller" and passing the component to it and handled the onChange with useEffect inside the component. please let me know if there is a better approach

**Comment by bluebill1049** [comment]
> Got it fixed using "Controller" and passing the component to it and handled the onChange with useEffect inside the component. please let me know if there is a better approach

can you share the code for the above?
